### ISSUE TITLE
ZTP switches claim 192.168.1.1 during autoinstall DHCP when using in-band ports

### ISSUE THREAD

**Issue by austind** [convergeone/freeztp#90]
**Edit**: at first, this looked like it only affected a certain IOS version. I have now observed this behavior on multiple versions. Having a hard time getting consistent results.

Running FreeZTP 1.5 with clean known-working config. has worked on many cat3k and cat9k deployments.
Hardware tested: C9200-48P
isc-dhcp-server: 4.3.5-3+deb9u2

When an affected device finishes bootstrapping, its shows expected console output:
```
         --- System Configuration Dialog ---

Would you like to enter the initial configuration dialog? [yes/no]:
```

FreeZTP log shows nothing.

`/var/log/syslog` shows:

```
Jun 29 08:33:24 raspberrypi dhcpcd[502]: eth1: hardware address 64:3a:ea:b3:5c:c7 claims 192.168.1.1
Jun 29 08:33:25 raspberrypi dhcpd[1377]: DHCPOFFER on 192.168.1.172 to 64:3a:ea:b3:5c:c7 (Switch) via eth1
Jun 29 08:33:26 raspberrypi dhcpcd[502]: eth1: hardware address 64:3a:ea:b3:5c:c7 claims 192.168.1.1
Jun 29 08:33:26 raspberrypi dhcpcd[502]: eth1: hardware address 64:3a:ea:b3:5c:c7 claims 192.168.1.1
Jun 29 08:33:27 raspberrypi dhcpd[1377]: DHCPDISCOVER from 64:3a:ea:b3:5c:c7 (Switch) via eth1
Jun 29 08:33:27 raspberrypi dhcpd[1377]: DHCPOFFER on 192.168.1.172 to 64:3a:ea:b3:5c:c7 (Switch) via eth1
Jun 29 08:33:31 raspberrypi dhcpd[1377]: DHCPDISCOVER from 64:3a:ea:b3:5c:c7 (Switch) via eth1
Jun 29 08:33:31 raspberrypi dhcpd[1377]: DHCPOFFER on 192.168.1.172 to 64:3a:ea:b3:5c:c7 (Switch) via eth1
Jun 29 08:33:39 raspberrypi dhcpd[1377]: DHCPDISCOVER from 64:3a:ea:b3:5c:c7 (Switch) via eth1
Jun 29 08:33:39 raspberrypi dhcpcd[502]: eth1: hardware address 64:3a:ea:b3:5c:c7 claims 192.168.1.1
Jun 29 08:33:40 raspberrypi dhcpd[1377]: DHCPOFFER on 192.168.1.176 to 64:3a:ea:b3:5c:c7 (Switch) via eth1
Jun 29 08:33:41 raspberrypi dhcpcd[502]: eth1: hardware address 64:3a:ea:b3:5c:c7 claims 192.168.1.1
Jun 29 08:33:41 raspberrypi dhcpcd[502]: eth1: hardware address 64:3a:ea:b3:5c:c7 claims 192.168.1.1
Jun 29 08:33:42 raspberrypi dhcpd[1377]: DHCPDISCOVER from 64:3a:ea:b3:5c:c7 (Switch) via eth1
Jun 29 08:33:42 raspberrypi dhcpd[1377]: DHCPOFFER on 192.168.1.176 to 64:3a:ea:b3:5c:c7 (Switch) via eth1
```

these messages repeat for other hardware addresses tested. They send discovers, isc-dhcp-server responds with an offer, then they claim the FreeZTP address (192.168.1.1) instead and autoinstall/ZTP stalls.

Found no issues in the ZTP config or generated /etc/dhcp/dhcpd.conf.

During one test, reverting to 16.12.04 solved issue. ZTP worked as expected. On another test on the same C9200 and IOS version, I ran into the same issue.

Quitting autoinstall and setting int vlan1 to dhcp obtains a normal lease from FreeZTP.

**Comment by pschapman** [comment]
Hi -

Is the switch able to obtain IP from another DHCP server? (e.g. Microsoft)

**Comment by austind** [comment]
> Hi -
> 
> Is the switch able to obtain IP from another DHCP server? (e.g. Microsoft)

Quitting autoinstall & setting int vlan1 to dhcp obtains a proper lease from FreeZTP.

Most recent batch doing the same thing on 16.12.04

**Comment by austind** [comment]
Changing from in-band ports to management port works consistently across IOS versions and switches.

Same switch, ZTP & DHCP config on in-band ports fails as described above.

Unclear how or why it ever worked on in-band ports, which I've been using up until now for all ZTP.

**Comment by pschapman** [comment]
Hi -

I just completed a deployment of (300) C9200-48P, (60) C9200-24P, and (50) C1000-16P.  I found that if uplinks came online before DHCP was available (and ZTP by extension), that the switches would kill the autoinstall process after about 15 minutes.  This would result in the VLAN 1 SVI being disabled and the switch would not be remotely manageable.

To avoid this, we intentionally left uplinks offline until we knew that the new site core was able to reach the ZTP server.  This would consistently prevent autoinstall from timing out and allow us to use the in-band connections for configuration.

(FreeZTP version 1.5.1 on Ubuntu 20 (all current patches) hosted on VMware)

PSC
